src/output.rs

```diff
@@ -0,0 +1,42 @@
+use num_format::{CustomFormat, Grouping};
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum NumberFormatStyle {
```

Comment:
Based on the feedback on `FormattableStats` this enum should be available in the library, and not just the binary, and should have `#[non_exhaustive]` so that we can add new formats without a major version bump.  What do you think about adding a `SystemLocale` variant that uses the `with-system-locale` feature in `num-format`?
```suggestion
#[non_exhaustive]
pub enum NumberFormatStyle {
```

Reply:
Good idea! I like the suggestion for `#[non_exhaustive]`.

About adding a system locale feature, I just want to reference the description from `num_format` and make sure it's ok:

> Since this type requires several dependencies (especially on Windows), it is behind a feature flag. To use it, include num-format = { version = "0.4", features = ["with-system-locale"] } in your Cargo.toml. Additionally, on Windows (but only on Windows), using SystemLocale requires Clang 3.9 or higher.

From what I can tell (not 100% sure on this part), it looks like this would add the following dependencies:
 - `num-format-windows`
 - `widestring`
 - `winapi/winnls`

Figured I would check :smile:

Reply:
Yeah that's totally fine if it adds those dependencies